packages/astro/src/cli/add/index.ts

```diff
@@ -848,25 +848,31 @@ async function updateTSConfig(
 		return UpdateResult.none;
 	}
 
-	const inputConfig = loadTSConfig(cwd, false);
-	const configFileName = inputConfig.exists ? inputConfig.path.split('/').pop() : 'tsconfig.json';
+	let inputConfig = await loadTSConfig(cwd);
+	let inputConfigText = '';
 
-	if (inputConfig.reason === 'invalid-config') {
+	if (inputConfig === 'invalid-config' || inputConfig === 'unknown-error') {
 		return UpdateResult.failure;
-	}
-
-	if (inputConfig.reason === 'not-found') {
+	} else if (inputConfig === 'missing-config') {
 		logger.debug('add', "Couldn't find tsconfig.json or jsconfig.json, generating one");
+		inputConfig = {
+			tsconfig: defaultTSConfig,
+			tsconfigFile: path.join(cwd, 'tsconfig.json'),
+			rawConfig: { tsconfig: defaultTSConfig, tsconfigFile: path.join(cwd, 'tsconfig.json') },
+		};
+	} else {
+		inputConfigText = JSON.stringify(inputConfig.rawConfig.tsconfig, null, 2);
 	}
 
+	const configFileName = inputConfig.tsconfigFile.split('/').pop();
```

Comment:
```suggestion
	const configFileName = inputConfig.tsconfigFile.split(path.sep).pop();
```

Looks like the bug existed before, but we might need to use proper OS separator, or use `path.basename` directly.